File: spring-best/src/main/java/com/best/springbest/model/ProductColor.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "product_color")
public class ProductColor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id",nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "color_id",nullable = false)
    private Color color;

    private String image;

}

File: BEST-App-main/lib/model/product_color.dart
class ProductColor {
  final int id;
  final int productId;
  final int colorId;
  final String image;

  ProductColor(this.id, this.productId,this.colorId, this.image);

  static Map<String, dynamic> toJson(ProductColor colorImage) {
    return {'id': colorImage.id, 'product_id': colorImage.productId,'color_id': colorImage.colorId, 'image': colorImage.image};
  }
}
